File: java_integration/app/src/main/java/com/firewok/example/java_integration/feedintegration/activity/ActivityIntegrationActivity.java
package com.firewok.example.java_integration.feedintegration.activity;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.MenuItem;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import com.firework.common.feed.FeedResource;
import com.firewok.example.java_integration.R;
import com.firewok.example.java_integration.databinding.ActivityActivityIntegrationBinding;
import com.firework.videofeed.FwVideoFeedView;
import com.firework.viewoptions.BaseOption;
import com.firework.viewoptions.ViewOptions;

public class ActivityIntegrationActivity extends AppCompatActivity {
    private ActivityActivityIntegrationBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityActivityIntegrationBinding.inflate(LayoutInflater.from(this));
        setContentView(binding.getRoot());
        
        setTitle(R.string.activity_screen_title);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setTitle(getString(R.string.activity_screen_title));
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setDisplayShowHomeEnabled(true);
        }
        
        initVideoFeedView();
    }

    private void initVideoFeedView() {
        FwVideoFeedView videoFeedView = binding.fwVideoFeedView;

        ViewOptions viewOptions = new ViewOptions.Builder().baseOption(new BaseOption.Builder().feedResource(FeedResource.Discovery.INSTANCE).build()).build();


        videoFeedView.init(viewOptions);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onDestroy() {
        binding.fwVideoFeedView.destroy();
        super.onDestroy();
    }

    public static Intent getIntent(Context context) {
        return new Intent(context, ActivityIntegrationActivity.class);
    }
} 

File: feed_integration/app/src/main/java/com/firework/example/feedintegration/activity/ActivityIntegrationActivity.kt
package com.firework.example.feedintegration.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import com.firework.common.feed.FeedResource
import com.firework.example.feedintegration.R
import com.firework.example.feedintegration.databinding.ActivityActivityIntegrationBinding
import com.firework.viewoptions.baseOptions
import com.firework.viewoptions.viewOptions

class ActivityIntegrationActivity : AppCompatActivity() {
    private lateinit var binding: ActivityActivityIntegrationBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityActivityIntegrationBinding.inflate(LayoutInflater.from(this))
        setContentView(binding.root)
        setTitle(R.string.activity_screen_title)
        supportActionBar?.title = getString(R.string.activity_screen_title)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setDisplayShowHomeEnabled(true)
        initVideoFeedView()
    }

    private fun initVideoFeedView() {
        val videoFeedView = binding.fwVideoFeedView

        val viewOptions =
            viewOptions {
                baseOptions {
                    feedResource(FeedResource.Discovery)
                }
            }

        videoFeedView.init(viewOptions)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onDestroy() {
        binding.fwVideoFeedView.destroy()
        super.onDestroy()
    }

    companion object {
        fun intent(context: Context) = Intent(context, ActivityIntegrationActivity::class.java)
    }
}
